agent/handlers/types/v2/stats_response.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package v2
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/engine/dockerstate"
+	"github.com/aws/amazon-ecs-agent/agent/stats"
+	"github.com/cihub/seelog"
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/pkg/errors"
+)
+
+// NewTaskStatsResponse returns a new task stats response object
+func NewTaskStatsResponse(taskARN string,
+	state dockerstate.TaskEngineState,
+	statsEngine stats.Engine) (map[string]*docker.Stats, error) {
+
+	containerMap, ok := state.ContainerMapByArn(taskARN)
+	if !ok {
+		return nil, errors.Errorf(
+			"v2 task stats response: unable to lookup containers for task %s",
+			taskARN)
+	}
+
+	resp := make(map[string]*docker.Stats)
+	for _, dockerContainer := range containerMap {
+		containerID := dockerContainer.DockerID
+		dockerStats, err := statsEngine.ContainerDockerStats(taskARN, containerID)
+		if err != nil {
+			seelog.Warnf("V2 task stats response: Unable to get stats for container '%s' for task '%s': %v",
+				containerID, taskARN, err)
+			resp[containerID] = nil
```

Comment:
Do you really need these two lines? Won't it just work without 42 and 43?

Reply:
Can you please elaborate?

Reply:
You can get rid of line 42 and 43, as the dockerStats will be nil if the err != nil, so this should work?
```
dockerStats, err := statsEngine.ContainerDockerStats(taskARN, containerID)
if err != nil {
     seelog.Warnf("V2 task stats response: Unable to get stats for container '%s' for task '%s': %v", 
        containerID, taskARN, err)
}
resp[containerID] = dockerStats
```

Reply:
i don't want to rely on the output of `statsEngine.ContainerDockerStats` for that (future bug proofing)